license-checker/npm/action.yml

```diff
@@ -0,0 +1,41 @@
+name: Check NPM License Compliance
+description: Check NPM dependencies for license compliance using license-checker
+
+inputs:
+  subdirectory:
+    description: The subdirectory to check dependencies
+    default: .
+    required: false
+  allowed_licenses:
+    description: Allowed license types (semicolon separated)
+    default: MIT;Apache;ISC;BSD;Python;CC;Public Domain;BlueOak;Unlicense
+    required: false
+  clarifications_file:
+    description: File to clarify package license
+    default: ../.github/clarification_file_license.json
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-node@v4
+      with:
+        node-version-file: ${{ inputs.subdirectory }}/package.json
+    - name: Enable corepack
+      shell: bash
+      run: corepack enable
+    - name: Run yarn install
+      shell: bash
+      working-directory: ${{ inputs.subdirectory }}
+      run: yarn install
+    - name: Install license-checker
+      shell: bash
+      run: npm install -g license-checker-rseidelsohn
+    - name: Check for License Compliance
+      shell: bash
+      working-directory: ${{ inputs.subdirectory }}
+      run: |
+        license-checker-rseidelsohn --excludePrivatePackages --production \
+          --clarificationsFile ${{ inputs.clarifications_file }} \
```

Comment:
related to: https://github.com/reearth/actions/pull/13/files#diff-41472c9976bcaed328a033186c584b48ee078000cbbc09b708ef65badc192903R15

what is the purpose of having it?
I see one example in https://github.com/RSeidelsohn/license-checker-rseidelsohn/blob/master/clarificationExample.json but not sure about the purpose. Can you enlighten me on this?